.storybook/preview.js

```diff
@@ -67,6 +68,22 @@ export const parameters = {
     // https://github.com/storybookjs/storybook/blob/v6.0.21/addons/docs/docs/docspage.md#inline-stories-vs-iframe-stories
     inlineStories: true,
     prepareForInline: (storyFn) => htmlToReactParser.parse(storyFn()),
+    transformSource(src, storyContext) {
+      try {
+        const storyFunction = storyContext.getOriginal();
+        const rendered = storyFunction(storyContext.args);
+        // The twing/source-inputs-loader.js file makes it so that whenever twig templates are rendered,
+        // the arguments and input path are stored in the window.__twig_inputs__ variable.
+        // __twig_inputs__ is a map between the output HTML and and objects with the arguments and input paths
+        // Here, since we have the rendered HTML, we can look up what the arguments and path were
+        // that correspond to that output
```

Comment:
This is really interesting! Clever!

Reply:
Thanks! This is the second approach, the first approach was to have the twig function return an object like this:

```js
{
  toString() {
    return '...html...'
  },
  path: '...',
  args: { ... }
}
```

which I think is a nicer solution, and it didn't have the limitation of not working for the stories which use useEffect / other hooks. But then I realized it breaks for the canvas page, because the canvas page has a typeof check and it doesn't allow non-strings (even though an object with a toString method can be implicitly coerced into a string)